### DISCUSSION TITLE
Multiple Domain Setup With Isso's Internal Server

### DISCUSSION THREAD

**Discussion by madranet** [posativ/isso#783] [Q&A]
The docs give a method using Gunicorn server for a multiple domain setup:

```
$ export ISSO_SETTINGS="/etc/isso.d/foo.example.cfg;/etc/isso.d/other.bar.cfg"
$ gunicorn isso.dispatch -b localhost:8080
```

I'm trying to get a multiple domain setup working with Isso's internal server, but not having much luck.

I have my two site config files saved in `/etc/isso.d/available/` and aliased to `/etc/isso.d/enabled/` 

I've got a `systemctl service` defined for starting Isso, that contains the line:

```
ExecStart="/path/to/isso -c /etc/isso.d/enabled/domain1.cfg;/etc/isso.d/enabled/domain2.cfg
```
When I run the service Isso fails to start.  However, using either of:

```
ExecStart="/path/to/isso -c /etc/isso.d/enabled/domain1.cfg
```
or:

```
ExecStart="/path/to/isso -c /etc/isso.d/enabled/domain2.cfg
```
it works fine. So, obviously the problem lies in trying to load the two `cfg` files chained with a semicolon [as per the Gunicorn example].

I've also tried doing it the `nginx` way with:

```
ExecStart="/path/to/isso -c /etc/isso.d/enabled/*
```

and 

```
ExecStart="/path/to/isso -c /etc/isso.d/
```

but neither of those worked either. 

So, how do I run multiple domains using the built-in server?

**Comment by madranet** [comment]
Well. This is really odd. 

I've temporarily commented out references to the `domain2.cfg` file in the `ExecStart` command, until I figure out where I'm going wrong. But Isso still seems to be working for both `domain1.com` and `domain2.com`.  The only thing that seems not quite right is that, in the notification emails I receive when someone comments on `domain2.com`, the link to the comment is constructed as if it was on `domain1.com` --yet the links to Delete/Activate the comment point to the correct `domain2.com` domain:

```
tester <tester@test.com> wrote:

blah blah reply

IP address: XX.XXX.XXX.XXX
Link to comment: http://domain1.com/some-post/#isso-88

---
Delete comment: https://domain2.com/isso/id/88/delete/xxxxxxxxxxxxxxxxxxxxx
Activate comment: https://domain2.com/isso/id/88/activate/xxxxxxxxxxxxxxxxxxxxx
```

I'm glad it seems to be working [ish], but I'm still confused as hell as to how I'm meant to se this up properly???

**Comment by ix5** [comment]
Did you eventually figure this out? The documentation on multiple sub-URIs should have improved in the meantime.

**Comment by ix5** [comment]
@madranet would you mind verifying whether this is still an issue?
I vaguely recall you managed to set up Isso correctly after me helping you in another PR.

**Reply by madranet** [reply]
>>@madranet would you mind verifying whether this is still an issue?

Sorry. I can't help you.  Like you say, it was a good while ago, so I can't remember what the final outcome was. 

And, as I'm currently only running Isso on a single domain, that kind of suggests I never did get to the bottom of things.
